### PR TITLE
(BIDS-1542) remove all the asterisks form the sql selects

### PR DESCRIPTION
**Pull request by MauserBitfly**

### CODE REVIEW

**Review by recy21 — APPROVED** [review]

File: services/services.go [hunk 1]
```diff
@@ -178,7 +184,7 @@ func getRelaysPageData() (*types.RelaysResp, error) {
 					limit 1
 				) as latest_slot
 			from (
-				select * 
+				builder_pubkey
```
Comment (D13ce) [inline services/services.go:187]: The `select` statement is still required.

Also, at least the `tag_id` has to be selected too as it is used to group this properly, see
https://github.com/gobitfly/eth2-beaconchain-explorer/blob/4887c3412af6f61c937a9d70d2ab363c6e8324eb/services/services.go#L191

So at least something like this:

```suggestion
				select builder_pubkey, tag_id
```

File: services/services.go [hunk 2]
```diff
@@ -240,7 +246,7 @@ func getRelaysPageData() (*types.RelaysResp, error) {
 			validators.validatorindex as proposer,
 			encode(exec_extra_data, 'hex') as block_extra_data
 		from (
-			select * 
+			select value, block_slot, builder_pubkey, proposer_fee_recipient, block_root, tag_id
```
Comment (D13ce) [inline services/services.go:249]: This is requiring `proposer_pubkey` as it is used for the `left join`
https://github.com/gobitfly/eth2-beaconchain-explorer/blob/4887c3412af6f61c937a9d70d2ab363c6e8324eb/services/services.go#L260
and the `group by`
https://github.com/gobitfly/eth2-beaconchain-explorer/blob/4887c3412af6f61c937a9d70d2ab363c6e8324eb/services/services.go#L264
below.

Like this:
```suggestion
			select value, block_slot, builder_pubkey, proposer_fee_recipient, block_root, tag_id, proposer_pubkey
```

**Review by D13ce — CHANGES_REQUESTED** [review]

File: exporter/rocketpool.go [hunk 3]
```diff
@@ -229,7 +229,7 @@ func (rp *RocketpoolExporter) InitDAOProposals() error {
 
 func (rp *RocketpoolExporter) InitDAOMembers() error {
 	dbRes := []RocketpoolDAOMember{}
-	err := rp.DB.Select(&dbRes, `select * from rocketpool_dao_members`)
+	err := rp.DB.Select(&dbRes, `select address, id, url, joined_time, last_proposal_time, rpl_bond_amount, unbonded_validator_count from rocketpool_dao_members`)
```
[the remark this answers is not in the document]
Comment (MauserBitfly) [reply exporter/rocketpool.go]: url removed from query

File: handlers/api.go [hunk 4]
```diff
@@ -296,7 +296,7 @@ func ApiEpoch(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	rows, err := db.ReaderDb.Query(`SELECT *, 
+	rows, err := db.ReaderDb.Query(`SELECT attestationscount, attesterslashingscount, averagevalidatorbalance, blockscount, depositscount, eligibleether, epoch, finalized, globalparticipationrate, proposerslashingscount, rewards_exported, totalvalidatorbalance, validatorscount, voluntaryexitscount, votedether, withdrawalcount,
```
[the remark this answers is not in the document]
Comment (MauserBitfly) [reply handlers/api.go:299]: added to the definitions

File: handlers/validatorRewards.go [hunk 5]
```diff
@@ -61,8 +61,9 @@ func ValidatorRewards(w http.ResponseWriter, r *http.Request) {
 
 func getUserRewardSubscriptions(uid uint64) [][]string {
 	var dbResp []types.Subscription
+
 	err := db.FrontendWriterDB.Select(&dbResp,
-		`select * from users_subscriptions where event_name=$1 AND user_id=$2`, strings.ToLower(utils.GetNetwork())+":"+string(types.TaxReportEventName), uid)
+		`select id, omitempty, user_id,omitempty, event_name, event_filter, last_sent_ts, last_sent_epoch, created_ts, created_epoch, event_threshold, unsubscribe_hash, internal_state from users_subscriptions where event_name=$1 AND user_id=$2`, strings.ToLower(utils.GetNetwork())+":"+string(types.TaxReportEventName), uid)
```
[the remark this answers is not in the document]
Comment (MauserBitfly) [reply handlers/validatorRewards.go]: removed BOTH omitempty

File: handlers/user.go [hunk 6]
```diff
@@ -1038,7 +1038,7 @@ func UserSubscriptionsData(w http.ResponseWriter, r *http.Request) {
 
 	subs := []types.Subscription{}
 	err = db.FrontendWriterDB.Select(&subs, `
-			SELECT *
+			SELECT id, omitempty, user_id,omitempty, event_name, event_filter, last_sent_ts, last_sent_epoch, created_ts, created_epoch, event_threshold, unsubscribe_hash, internal_state
```
[the remark this answers is not in the document]
Comment (MauserBitfly) [reply handlers/user.go]: removed BOTH omitempty ;-)

File: services/services.go (hunk 1 above)
Comment (MauserBitfly) [reply services/services.go]: added tag_id

File: services/services.go (hunk 2 above)
Comment (MauserBitfly) [reply services/services.go]: added proposer_pubkey

File: handlers/validatorRewards.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (D13ce) [reply handlers/validatorRewards.go]: Wow, I did not even catch the first one sitting right there :D

File: handlers/user.go (hunk 6 above)
[the remark this answers is not in the document]
Comment (D13ce) [reply handlers/user.go]: Jeah jeah :stuck_out_tongue_winking_eye:

File: exporter/rocketpool.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (D13ce) [reply exporter/rocketpool.go]: Okay, I have tested this now. The url is returned even-though thanks to the according struct field being tagged `url:"url"`.
=> Please re-add url to the query.

[the remark this answers is not in the document]
Comment (MauserBitfly) [reply exporter/rocketpool.go]: url back in the query

**Review by D13ce — APPROVED** [review]
utACK

**Pull request merged**